packages/features/config-context.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+
+import { ChildrenProps, ReactEvent } from '@packages/utils/react';
+import createCtx from '@packages/utils/createCtx';
+import { FormState } from '@packages/entities/config-modal';
+
+interface ConfigActions {
+  readonly onConfigChange: (event: ReactEvent) => void;
+}
+
+interface ConfigStates {
+  readonly notionApiKey: string;
+  readonly electionDatabaseId: string;
+  readonly resultsDatabaseId: string;
+}
+
+const [useConfigActions, ConfigActionsProvider] =
+  createCtx<ConfigActions>('ConfigActionsCtx');
+const [useConfigStates, ConfigStatesProvider] =
+  createCtx<ConfigStates>('ConfigStatesCtx');
+
+function ConfigProvider({ children }: ChildrenProps) {
+  const [formState, setFormState] = useState<FormState>({
+    notionApiKey: '',
+    electionDatabaseId: '',
+    resultsDatabaseId: '',
+  });
+
+  const onConfigChange = (event: ReactEvent) => {
+    setFormState({
+      ...formState,
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  const actions: ConfigActions = {
+    onConfigChange: (event) => onConfigChange(event),
```

Comment:
Can we simplify this?

```suggestion
    onConfigChange,
```

Reply:
lol,  of course! Dindn't notice

Reply:
done!